File: packages/core/dto/replicaset.go
package dto

type ReplicaSetRef struct {
	Namespace string `json:"namespace"`
	Name      string `json:"name"`
}

type ReplicaSet struct {
	Name           string
	Namespace      string
	Desired        int32
	Current        int32
	Ready          int32
	Age            string
	OwnerName      string
	CreatedAt      string
	OwnerKind      string
	Labels         map[string]string
	Annotations    map[string]string
	ManagedFields  []ManagedField
	Selector       string
	NodeSelector   string
	Images         []string
	ReplicasDetail string
	Tolerations    int
	Affinities     int
	PodStatus      string
}

type ReplicaSetSummary struct {
	Running int
	Pending int
}

File: packages/core/frontend/src/types/resources/replicaset.ts
import type { ManagedField } from "./shared";

export interface ReplicaSet {
  Name: string;
  Namespace: string;
  Desired: number;
  Current: number;
  Ready: number;
  Age: string;
  OwnerName: string;
  CreatedAt: string;
  OwnerKind: string;
  Labels: Record<string, string>;
  Annotations: Record<string, string>;
  ManagedFields: ManagedField[];
  Selector: Record<string, string>;
  NodeSelector: Record<string, string>;
  Images: string[];
  ReplicasDetail: string;
  Tolerations: number;
  Affinities: number;
  PodStatus: string;
  HealthStatus: string;
  HealthMessage: string;
}

export interface ReplicaSetSummary {
  Running: number;
  Pending: number;
}
